src/components/inputs/DigitInput.js

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+
+const DigitInput = ({ value, setValue, withPeriod = false, ...rest }) => {
+  const isValid = (reg, val) => {
+    return val.match(reg);
+  };
+
+  const handleChange = e => {
+    const val = e.target.value;
+
+    //allows to delete first char
+    if (val.length === 0) {
```

Comment:
Please do not treat this case  separately, it makes code harder to read. We can just use wildcard in the regex which will solve this problem for us.